core/store/src/db.rs

```diff
@@ -249,6 +250,12 @@ pub trait Database: Sync + Send {
         path: &std::path::Path,
         columns_to_keep: Option<&[DBCol]>,
     ) -> anyhow::Result<()>;
+
+    /// If this is a test database, return a copy of the entire database.
+    /// Otherwise return null.
+    fn copy_if_test(&self) -> Option<Arc<dyn Database>> {
```

Comment:
Please check if it is possible to hide under `#[cfg(test)]`.

Reply:
It's not, cfg(test) is crate-local.